classes/category/Collector.php

```diff
@@ -137,7 +139,9 @@ public function getQueryBuilder(): Builder
 
         $qb->when($this->publicationIds !== null, function ($query) {
             $query->whereIn('c.category_id', function ($query) {
-                $query->select('category_id')->from('publication_categories')->whereIn('publication_id', $this->publicationIds);
+                PublicationCategory::select('category_id')
+                ->withPublicationIds($this->publicationIds)
+                ->toBase();
```

Comment:
Let me know if I'm wrong, but I've just inspected the internals of Laravel and looks like this idea isn't going to work. Laravel isn't using the return of the callback on the `whereIn()`, so it's needed to use the `$query` that it provided you. Also, you're not returning anything on the function.

Reply:
@jonasraoni  You’re right! thank you for inspecting Laravel's internal the implementation had issues with how i was using the whereIn() method and the callback function. I am fixing it by using the `whereIn()` method without relying on a callback return and utilizing the `PublicationCategory` model to create a proper subquery.